presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java

```diff
@@ -429,7 +432,7 @@ public ReadContext init(
             ImmutableList.Builder<parquet.schema.Type> fields = ImmutableList.builder();
             for (HiveColumnHandle column : columns) {
                 if (!column.isPartitionKey() && column.getHiveColumnIndex() < messageType.getFieldCount()) {
-                    fields.add(messageType.getType(column.getName()));
+                    fields.add(getParquetType(configuration, column, messageType));
```

Comment:
There are two scenarios.

1) The Hive Way
This is what we have today.  The order of the column names in the Hive metastore is expected to match the exact order of the columns in the file (the column names in the Parquet files are completely ignored).  So if the metastore has `A:INT, B:VARCHAR, C:DOUBLE`, the files are expected to have `_:INT, _:VARCHAR, _:DOUBLE`.  If someone, executes `ALTER TABLE test_change CHANGE a x INT`, the metastore will have ``X:INT, B:VARCHAR, C:DOUBLE`, and the file is expected to have `_:INT, _:VARCHAR, _:DOUBLE`.  This means that Hive can rename columns without having to rewrite the files, but other system must write columns in an exact order declared in the Hive metastore.

2) The Parquet Way
FWIU, this is what Parquet users do.  The Parquet files can have any columns in any order, and the Hive metastore contains a list of column names in the table.  So if the metastore has `A:INT, B:VARCHAR, C:DOUBLE`, and the Parquet file has `Q:BOOLEAN, B:VARCHAR, A:INT, X:INT`, the user will see a table with Parquet columns `3,2,null` (`A:INT, B:VARCHAR, null`).  If someone, executes `ALTER TABLE test_change CHANGE a x INT`, the metastore will have `X:INT, B:VARCHAR, C:DOUBLE`, the user will see a table with parquet columns `4,2,null` (`X:INT, B:VARCHAR, null`).  This means that Hive column rename command is actually changing the column selected from the Parquet file, and does not actually rename the columns.  The downside is there is no way to rename columns without rewriting all of the data.

For this commit, when "The Parquet Way" is enabled, we need to map the Hive metastore column ordinals to the Parquet file column ordinals based on name.  For the example above, if we are reading `A`, Presto will look in the Hive metastore and see it is the first column in the list, then when we open the Parquet file, we see that it is the third column in file.  So when Presto asks for column `1`, this code must return data from Parquet column `3`.